### PR TITLE
sql-statements: use EBNF to render syntax diagrams for ADD, ALTER and ANALYZE statements

### PR DESCRIPTION
**Pull request by CharLotteiu**
### What is changed, added or deleted? (Required)

This PR changes the synopsis sections of ADD, ALTER and ANALYZE statements to use the ` ```ebnf+diagrm ` block, replacing the original image-based sqlgrams. It is a follow-up PR to https://github.com/pingcap/docs/issues/4389 and https://github.com/pingcap/docs/pull/4342. 

To avoid making one single PR too large, I will open 4-5 follow-up PRs to process sql-statements files in batches. And I will delete all image files in the last PR in case of any display issues.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.0 (TiDB 5.0 versions)
- [x] v4.0 (TiDB 4.0 versions)
- [x] v3.1 (TiDB 3.1 versions)
- [x] v3.0 (TiDB 3.0 versions)
- [x] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): https://pingcap.github.io/sqlgram/

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [x] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: expression-syntax.md [hunk 1]
```diff
@@ -17,18 +17,32 @@ aliases: ['/docs-cn/dev/expression-syntax/','/docs-cn/dev/reference/sql/language
...
+
+PredicateExpr ::=
+    BitExpr ( BetweenOrNotOp BitExpr 'AND' BitExpr )* ( InOrNotOp ( '(' ExpressionList ')' | SubSelect ) | LikeOrNotOp SimpleExpr LikeEscapeOpt | RegexpOrNotOp SimpleExpr )?
+
+BitExpr ::=
+    BitExpr ( ( '|' | '&' | '<<' | '>>' | '*' | '/' | '%' | 'DIV' | 'MOD' | '^' ) BitExpr | ( '+' | '-' ) ( BitExpr | "INTERVAL" Expression TimeUnit ) )
+|   SimpleExpr
+
+SimpleExpr ::=
+    SimpleIdent ( ( jss | juss ) stringLit )?
```
Comment (kennytm) [inline expression-syntax.md:34]:
```suggestion
    SimpleIdent ( ( '->' | '->>' ) stringLit )?
```

File: expression-syntax.md [hunk 2]
```diff
@@ -17,18 +17,32 @@ aliases: ['/docs-cn/dev/expression-syntax/','/docs-cn/dev/reference/sql/language
...
+|   FunctionCallKeyword
+|   FunctionCallNonKeyword
+|   FunctionCallGeneric
+|   SimpleExpr ( 'COLLATE' CollationName | pipes SimpleExpr )
+|   WindowFuncCall
+|   Literal
+|   paramMarker
+|   Variable
+|   SumExpr
+|   ( '!' | '~' | '-' | '+' | not2 | 'BINARY' ) SimpleExpr
```
Comment (kennytm) [inline expression-syntax.md:44]:
```suggestion
|   ( '!' | '~' | '-' | '+' | 'NOT' | 'BINARY' ) SimpleExpr
```

File: sql-statements/sql-statement-add-column.md [hunk 3]
```diff
@@ -10,21 +10,39 @@ aliases: ['/docs-cn/dev/sql-statements/sql-statement-add-column/','/docs-cn/dev/
...
+|   'COALESCE' 'PARTITION' NoWriteToBinLogAliasOpt NUM
+|   'DROP' ( ColumnKeywordOpt IfExists ColumnName RestrictOrCascadeOpt | 'PRIMARY' 'KEY' | 'PARTITION' IfExists PartitionNameList | ( KeyOrIndex IfExists | 'CHECK' ) Identifier | 'FOREIGN' 'KEY' IfExists Symbol )
+|   'EXCHANGE' 'PARTITION' Identifier 'WITH' 'TABLE' TableName WithValidationOpt
+|   ( 'IMPORT' | 'DISCARD' ) ( 'PARTITION' AllOrPartitionNameList )? 'TABLESPACE'
+|   'REORGANIZE' 'PARTITION' NoWriteToBinLogAliasOpt ReorganizePartitionRuleOpt
+|   'ORDER' 'BY' AlterOrderItem ( ',' AlterOrderItem )*
+|   ( 'DISABLE' | 'ENABLE' ) 'KEYS'
+|   ( 'MODIFY' ColumnKeywordOpt IfExists | 'CHANGE' ColumnKeywordOpt IfExists ColumnName ) ColumnDef ColumnPosition
+|   'ALTER' ( ColumnKeywordOpt ColumnName ( 'SET' 'DEFAULT' ( SignedLiteral | '(' Expression ')' ) | 'DROP' 'DEFAULT' ) | 'CHECK' Identifier EnforcedOrNot | 'INDEX' Identifier IndexInvisible )
+|   'RENAME' ( ( 'COLUMN' | KeyOrIndex ) Identifier 'TO' Identifier | ( 'TO' | EqOpt | 'AS' ) TableName )
```
Comment (kennytm) [inline sql-statements/sql-statement-add-column.md:32]: I'd prefer inlining all `EqOpt` as `'='?` on display

Comment (CharLotteiu) [reply sql-statements/sql-statement-add-column.md]: Updated in `21fdb00`.

**Comment by kennytm** [comment]
LGTM

**Review by TomShawn — APPROVED** [review]

**Comment by TomShawn** [comment]
@CharLotteiu Shall we remove the png files in this PR?

**Comment by CharLotteiu** [comment]
> @CharLotteiu Shall we remove the png files in this PR?

I prefer to remove all png files in the last PR because some of the png files might still be used in other md files. What do you think?

**Pull request merged**